src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java

```diff
@@ -618,7 +746,9 @@ protected Clock getClock() {
                 new ScalingExecutorBuilder(
                     AD_THREAD_POOL_NAME,
                     1,
-                    Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) / 4),
+                    // HCAD can be heavy after supporting 1 million entities.
+                    // Limit to use at most half of the processors.
+                    Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) / 2),
```

Comment:
The threadpool size is determined at the beginning of node creation.  I don't know how to replace it with a new threadpool with another size dynamically.  Could you tell me how to do that?

Reply:
Do we have a way for static config? something like the brazil config?
If yes, it's better to move it to the static config, otherwise, it's fine to keep the current code.

Reply:
We don't have a .config file :)